src/main/scala/viper/server/vsi/Cache.scala

```diff
@@ -156,28 +154,32 @@ abstract class Cache {
         val existing_entries = fileCache.getOrElse(concerning_hash, Nil)
         val updated_cacheEntries = new_entry :: existing_entries
 
-        fileCache(concerning_hash) = updated_cacheEntries
+        val updatedFileCache = fileCache + (concerning_hash -> updated_cacheEntries)
+        _cache = _cache + (file_key -> updatedFileCache)
         updated_cacheEntries
       case None =>
         //if file not in cache yet, create new map entry for it. Recall the function.
-        _cache += (file_key -> collection.mutable.Map[String, List[CacheEntry]]())
+        _cache = _cache + (file_key -> Map())
         update(file_key, key, dependencies, content)
     }
   }
 
   /** Resets the cache for a particular file.
     * */
   def forgetFile(file_key: String): Option[String] = {
-    _cache.remove(file_key) match {
+    val elem = _cache.get(file_key) match {
       case Some(_) => Some(file_key)
       case None => None
     }
+    _cache -= file_key
+    elem
```

Comment:
this function now behalves differently than before. Could it return `Unit` instead?

Reply:
I think it still behaves the same since it previously removed a cache entry and then returned the cache key if it existed. I also find it strange to return the key you gave to the method instead of the value, but I didn't want to change the behaviour.

Reply:
You're right. I got confused by `elem` and somehow missed that you ignored `get`'s result. Please rename the variable to `key`